lib/IRGen/GenReflection.cpp

```diff
@@ -1006,6 +1006,65 @@ void IRGenModule::emitBuiltinTypeMetadataRecord(CanType builtinType) {
   builder.emit();
 }
 
+class MultiPayloadEnumDescriptorBuilder : public ReflectionMetadataBuilder {
+  CanType type;
+  const FixedTypeInfo *ti;
+
+public:
+  MultiPayloadEnumDescriptorBuilder(IRGenModule &IGM,
+                                    const NominalTypeDecl *nominalDecl)
+    : ReflectionMetadataBuilder(IGM) {
+    type = nominalDecl->getDeclaredType()->getCanonicalType();
+    ti = &cast<FixedTypeInfo>(IGM.getTypeInfoForUnlowered(
+        nominalDecl->getDeclaredTypeInContext()->getCanonicalType()));
+  }
+
+  void layout() override {
+    // TODO: Don't try to emit MPE descriptors for unbound generic enums
+    // Related: getEnumImplStrategy() below asserts that `type` is not an unbound generic
+    // if (type is generic) {
+    //   return;
+    // }
+
+    addTypeRef(type, CanGenericSignature());
+
+    auto &strategy = getEnumImplStrategy(IGM, type);
```

Comment:
TODO: getEnumImplStrategy here is crashing, apparently because `type` is sometimes an unbound generic?  I don't believe I need to emit an MPE descriptor for unbound generics, so it should suffice to omit the MPE descriptor in that case.

Reply:
@aschwaighofer Can you help me figure out what `type` I should be using here?

Reply:
You will want something like `CanType getFormalTypeInContext(CanType abstractType)` -- it maps the interface types into the context of the type's generic signature.

Reply:
I copied that function from `GenEnum.cpp` and it seems to stop the crashing.
TODO: Share that function instead of copying it.

Reply:
TODO copied below; rest of this seems to be working.